### ISSUE TITLE
Suggestion for Improvement

### ISSUE THREAD

**Issue by Susanna-lehner** [kovzol/aris#23]
Dear GNU-Aris Team,

I have another suggestion to improve your program: It would be great if it allowed zooming in the window.

Kind regards,
Susanna

**Comment by kovzol** [comment]
Thanks, indeed! It is very annoying that zooming is not possible at the moment.

**Comment by swymbnsl** [comment]
Hey @kovzol 

I tried to implement zoom in QT ui of Aris but ran into an issue.

If we keep zooming in, the KeyGroup window starts overlapping the Burger Menu Button.

Now 1 possible fix could be to limit zooming in to a value that it doesn't break in the default window size (1200x720 px).

Another solution that seems better to me is redesigning the UI such that it can be made more responsive. 

Can you suggest me which way I can resolve the issue better?

Thank you

**Comment by kovzol** [comment]
Hi, thanks for the suggestion. Both make sense. For the long term, I think, redesigning the UI is a better idea, but for a quick fix you can try the first option too.

**Comment by swymbnsl** [comment]
Yes, it's better to fix it for a long term, so redesigning the UI makes sense.

I will come up with a design concept and share it soon. I also plan to include a few additional enhancements in the PR, such as changing the line type after line creation (premise to conclusion and vice versa), adding a language selector, and a few more.

**Comment by swymbnsl** [comment]
Hello @kovzol 

Here's the design I came up with

#### Light Mode:

<img width="1206" height="733" alt="Image" src="https://github.com/user-attachments/assets/4cfc4d47-fdff-4a86-bb0f-551685db9f90" />

#### Dark mode:

<img width="1206" height="733" alt="Image" src="https://github.com/user-attachments/assets/38183426-ff2f-423b-ab2a-02f2425c8e49" />


I added a `RowLayout` in `main.qml` which holds a sidebar and the `ProofArea`. So basically, instead of directly rendering the `burgerMenu` and `KeyGroup` on the `ProofArea`, they are now rendered in a dedicated sidebar that is beside the `ProofArea`. I also put the `KeyGroup` inside a `ScrollView` so that when zoomed in or when opened on smaller screens, all options remain accessible rather than being cut off. This new layout will solve that overlapping issue we were facing earlier

<img width="1196" height="365" alt="Image" src="https://github.com/user-attachments/assets/84b8abb0-889c-45ce-80ea-6f4b086bd130" />

Please let me know if I should continue with this design or change something. I can then implement zoom in the finalized UI.
